crates/biome_deserialize/src/impls.rs

```diff
@@ -0,0 +1,571 @@
+        None
+    }
+}
+
+impl Deserializable for NonZeroU64 {
+    fn deserialize(
+        value: impl DeserializableValue,
+        diagnostics: &mut Vec<DeserializationDiagnostic>,
+    ) -> Option<Self> {
+        let range = value.range();
+        let value = TokenNumber::deserialize(value, diagnostics)?;
+        if let Ok(value) = value.text().parse::<Self>() {
+            return Some(value);
+        }
+        let diagnostic = DeserializationDiagnostic::new_out_of_bound_integer(
+            Self::MIN.get(),
+            Self::MAX.get(),
+            range,
+        );
+        diagnostics.push(diagnostic);
+        None
+    }
+}
+
+impl Deserializable for TokenText {
+    fn deserialize(
+        value: impl DeserializableValue,
+        diagnostics: &mut Vec<DeserializationDiagnostic>,
+    ) -> Option<Self> {
+        struct Visitor;
+        impl DeserializationVisitor for Visitor {
+            type Output = TokenText;
+            const EXPECTED_TYPE: ExpectedType = ExpectedType::STR;
+            fn visit_str(
+                self,
+                value: TokenText,
+                _range: TextRange,
+                _diagnostics: &mut Vec<DeserializationDiagnostic>,
+            ) -> Option<Self::Output> {
+                Some(value)
+            }
+        }
+        value.deserialize(Visitor, diagnostics)
+    }
+}
+
+impl Deserializable for String {
+    fn deserialize(
+        value: impl DeserializableValue,
+        diagnostics: &mut Vec<DeserializationDiagnostic>,
+    ) -> Option<Self> {
+        TokenText::deserialize(value, diagnostics).map(|value| value.to_string())
+    }
+}
+
+impl Deserializable for PathBuf {
+    fn deserialize(
+        value: impl DeserializableValue,
+        diagnostics: &mut Vec<DeserializationDiagnostic>,
+    ) -> Option<Self> {
+        String::deserialize(value, diagnostics).map(PathBuf::from)
+    }
+}
+
+impl<T: Deserializable> Deserializable for Vec<T> {
+    fn deserialize(
+        value: impl DeserializableValue,
+        diagnostics: &mut Vec<DeserializationDiagnostic>,
+    ) -> Option<Self> {
+        struct Visitor<T>(PhantomData<T>);
+        impl<T: Deserializable> DeserializationVisitor for Visitor<T> {
+            type Output = Vec<T>;
+            const EXPECTED_TYPE: ExpectedType = ExpectedType::ARRAY;
+            fn visit_array(
+                self,
+                values: impl Iterator<Item = impl DeserializableValue>,
+                _range: TextRange,
+                diagnostics: &mut Vec<DeserializationDiagnostic>,
+            ) -> Option<Self::Output> {
+                Some(
+                    values
+                        .filter_map(|value| Deserializable::deserialize(value, diagnostics))
+                        .collect(),
+                )
+            }
+        }
+        value.deserialize(Visitor(PhantomData), diagnostics)
+    }
+}
+
+impl<T: Deserializable + Eq + Hash, S: BuildHasher + Default> Deserializable for HashSet<T, S> {
```

Comment:
While this is useful, `HashSet` and `Vec` **don't** guarantee insertion order, so we shouldn't provide functionalities for these types. In fact, internally, we use `StringSet` for all the arrays we have, which is a tiny wrapper for `IndexSet`. 

This will bite us if/when we will implement `Serialize`

Reply:
> [...] Vec don't guarantee insertion order

`Vec` is by design an ordered sequence?

Reply:
Does `Vec` guarantee insertion order? I was under the assumption that it does not.

Reply:
It depends on how you add items. If you use `push`, data is always appended. Thus, the order is preserved.